Content.Server/_Floof/Preferences/FavoriteDrinkSystem.cs

```diff
@@ -0,0 +1,26 @@
+using Content.Server.GameTicking;
+using FavoriteDrinkComponent = Content.Server._Floof.Preferences.FavoriteDrinkComponent;
+
+
+namespace Content.Server._Floof.Traits;
+
+public sealed class FavoriteDrinkSystem : EntitySystem
+{
+    public override void Initialize()
+    {
+        SubscribeLocalEvent<PlayerSpawnCompleteEvent>(OnPlayerSpawnComplete);
+    }
+
+    private void OnPlayerSpawnComplete(PlayerSpawnCompleteEvent args)
+    {
+        if (args.Profile.FavoriteDrink is not { } favoriteDrink)
+            return;
+
+        var comp = new FavoriteDrinkComponent()
+        {
+            ReagentId = favoriteDrink
+        };
+
+        AddComp(args.Mob, comp);
```

Comment:
Cleaner way to do this:
```suggestion
        var comp = EnsureComp<FavoriteDrinkComponent>(args.Mob);
        comp.ReagentId = favoriteDrink;
```
Also note that this entire system could be rendered unnecessary as you can fetch HumanoidCharacterProfile.FavoriteDrink directly from HumanoidAppearenceComponent, and use FavoriteDrinkComponent as a fallback

Reply:
took your suggestion but i decided to leave the method in because i think it's confusing to have multiple sources of truth for this (for one thing making it impossible or confusing to remove the favorite drink preference in-game with VV, and HumanoidAppearanceComponent is not copied over to polymorphs for good reason, while it's more probable that FavoriteDrinkComponent will be allowed to copy over in the future)